java/arcs/core/host/ParticleStateMachine.kt

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.core.host
+
+import arcs.core.entity.Handle
+import arcs.core.host.ParticleEvent.CreateEvent
+import arcs.core.host.ParticleEvent.FailedEvent
+import arcs.core.host.ParticleEvent.MaxFailedEvent
+import arcs.core.host.ParticleEvent.ReadyEvent
+import arcs.core.host.ParticleEvent.StartEvent
+import arcs.core.host.ParticleEvent.StopEvent
+import arcs.core.host.ParticleState.Created
+import arcs.core.host.ParticleState.Failed
+import arcs.core.host.ParticleState.Failed_NeverStarted
+import arcs.core.host.ParticleState.Instantiated
+import arcs.core.host.ParticleState.MaxFailed
+import arcs.core.host.ParticleState.Ready
+import arcs.core.host.ParticleState.Started
+import arcs.core.host.ParticleState.Stopped
+import arcs.core.host.api.Particle
+import arcs.core.util.StateMachine
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+
+enum class ParticleEvent {
+    StartEvent,
+    StopEvent,
+    FailedEvent,
+    MaxFailedEvent,
+    CreateEvent,
+    ReadyEvent
```

Comment:
I don't think we need the `Event` suffix on all of these, since they're already nicely namespaced under `ParticleEvent`

Also, KDoc for the overall Enum class would be good.